### DISCUSSION TITLE
Rotating Ptfm orientation for FAST.Farm

### DISCUSSION THREAD

**Discussion by MYMahfouz** [OpenFAST/openfast#1902] [Q&A]
Dear all,

For FAST.Farm simulations I rotate the entire wind farm. However, i did not rotate the floating platform itself before. I have a method to rotate the platform when I rotate the entire wind farm but I am not sure it is correct.
- First: set `PtfmRefztRot` to the angle of rotation of the wind farm which is the same angle of the wind direction. Which reference frame does this value follow? For example if the rotation is to achieve wind direction of 10 degrees (positive anticlockwise relative to the x-axis), would I set  `PtfmRefztRot` to positive 10 degrees or negative 10 degrees?
- To account for Morison elements, I believe I should also rotate the joints definition in section `MEMBER JOINTS` in the HydroDyn file. is this correct?
- The `WaveDir` in HydroDyn input file would remain at zero

Are these steps accurate? Am I missing something?

Thank you.

Best regards,
Youssef

**Comment by jjonkman** [comment]
Dear @MYMahfouz,

I agree that if you are rotating the wind farm in FAST.Farm so as to use the same wind input for different wind directions, it is necessary to also rotate the offshore support structure.  I'm not sure what convention you are using for the wind direction (because presumably you are using the zero-degree wind direction in FAST.Farm), but I would think that you should rotate the offshore support structure by the same amount as the wind farm.

In addition to rotating the joints of the Morison elements, you should also rotate the joints in SubDyn (if enabled) and the fairlead/anchor positions in the mooring module.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman ,
Thanks for your reply.

Yes, I rotate the anchors and the fairlead positions in all my previous work. But up to this point I was not rotating the floating structure excitation forces.

I believe I understand the rotation of the Morison elements, and I found this [document ](https://www.nrel.gov/docs/fy20osti/76822.pdf) for the usage of `PtfmRefztRot`.

I have a question about setting `PtfmRefztRot`:
If I set `PtfmRefztRot = 30` degrees and `WaveDir = 0`, what angle would be read from the  wave-excitation coefficients from the hydrodynamic tables? 
According to the first line after equation 43 in the document above this value should be `-30`, is my understanding correct?

Thank you.

Best regards,

**Reply by jjonkman** [reply]
Yes, that is correct.
